client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -363,19 +378,50 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       }
     }
 
+    def processResponseWithCache(
+        context: RpcCallContext,
+        partitionLocations: Array[PartitionLocation] = Array.empty): Unit = {
+      if (context.isInstanceOf[LocalNettyRpcCallContext]) {
+        context.reply(RegisterShuffleResponse(
+          StatusCode.SUCCESS,
+          getInitialLocs(partitionLocations)))
+      } else {
+        val cachedMsg = getRegisterShuffleResponseRpcCache.get(
+          shuffleId,
+          new Callable[ByteBuffer]() {
+            override def call(): ByteBuffer = {
+              val returnedMsg =
+                RegisterShuffleResponse(StatusCode.SUCCESS, getInitialLocs(partitionLocations))
+              context.asInstanceOf[RemoteNettyRpcCallContext].nettyEnv.serialize(returnedMsg)
+            }
+          })
+        context.asInstanceOf[RemoteNettyRpcCallContext].callback.onSuccess(cachedMsg)
+      }
+    }
+
+    def getInitialLocs(partitionLocations: Array[PartitionLocation]): Array[PartitionLocation] = {
+      var initialLocs: Array[PartitionLocation] = partitionLocations
+      if (initialLocs.nonEmpty) {
```

Comment:
Shouldn't this not be `isEmpty` ?
If yes, why aren't the current tests not catching this ? Can you add a test to catch this ?